FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +23,46 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case .failure:
+				completion(.failure(Error.connectivity))
+			case let .success((data, httpResponse)):
+				guard httpResponse.statusCode == Constants.successSatusCode,
+				      let images = try? JSONDecoder().decode(FeedLoaderResponse.self, from: data) else {
+					completion(.failure(Error.invalidData))
+					return
+				}
+
+				completion(.success(images.items.feedImageItems()))
```

Comment:
Consider moving all this logic to a helper function, to make the `load` function shorter and easier to read. Suggestion:
```suggestion
				completion(RemoteFeedLoader.map(data, from: httpResponse))
```